### PR TITLE
set failover connector to alpha

### LINKED ISSUE #32094: [connector/failover] Failover connector erroneously flips back to lower priority pipelines
### Component(s)

connector/failover

## Description

The failover connector periodically retries higher priority pipelines that have failed, so that it can reinstate them as the stable pipeline should they start working again. We observe however that when it does so, it then reinstates the _lower_ priority pipeline, even when the higher priority pipeline is working.

## Steps to Reproduce

1. Create a pipeline with two exporters, connected with a failover connector.
2. Establish a job inserting logs so we can observe the output:

```bash
while true; do sleep 1; echo 'test' > logs; done
```

3. Start two listeners on each of the receiving ports:

```bash
nc -l 127.0.0.1 4278 # the high priority exporter
nc -l 127.0.0.1 4279 # the low priority exporter
```

4. Observe that logs correctly flow to the high priority exporter
5. Shut down the high priority exporter
6. Observe that logs correctly flow to the low priority exporter
7. Restart the high priority exporter

```bash
nc -l 127.0.0.1 4278 # the high priority exporter
```

8. Observe that logs correctly flow to the high priority exporter, but then after a few seconds fall back to the low priority exporter (and that it begins to flip flop back and forth)

## Expected Result

The logs should be stably redirected to the high priority exporter once it comes back online

## Actual Result

The logs flip flop between the high and low priority exporters

## Investigation

Adding a bit more logging around pipeline decisions finds that the lower priority pipeline is being re-inserted at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/connector/failoverconnector/internal/state/pipeline_selector.go#L105-L107 

This is because the loop terminates for pipelines after, _but including_ the current pipeline (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/connector/failoverconnector/internal/state/pipeline_selector.go#L96-L98). This means that while the lower priority pipeline is active, it creates a job that makes it active again, even if we select a higher priority pipeline.

### Collector version

master (failover connector isn't released yet)

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### OpenTelemetry Collector configuration

```yaml
receivers:
  namedpipe:
    path: ./logs

exporters:
  syslog/a:
    endpoint: localhost
    port: 4278
    retry_on_failure:
      enabled: false
    tls:
      insecure: true
  syslog/b:
    endpoint: localhost
    port: 4279
    retry_on_failure:
      enabled: false
    tls:
      insecure: true

connectors:
  failover:
    retry_interval: 10s
    retry_gap: 3s
    priority_levels:
      - [logs/a]
      - [logs/b]

service:
  pipelines:
    logs:
      receivers: [namedpipe]
      exporters: [failover]
    logs/a:
      receivers: [failover]
      exporters: [syslog/a]
    logs/b:
      receivers: [failover]
      exporters: [syslog/b]
```

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by akats7**
**Description:** <Describe what has changed.>

This PR adds the failover connector to the contrib distro and moves the component to alpha state as all MVP functionality has been put in place. This PR also adds a bug fix that caused the pipeline selector to continue switching between the stable and stable + 1 index after a new stable index has been established.

**Link to tracking Issue:** <Issue number if applicable>

Resolves #32094

**Testing:** <Describe what testing was performed and which tests were added.>

Additional test case added to check current index after stable check

### CODE REVIEW

File: cmd/otelcontribcol/go.mod [hunk 1]
```diff
@@ -8,6 +8,8 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector v0.97.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector v0.97.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/connector/exceptionsconnector v0.97.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/connector/failoverconnector v0.97.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/connector/routingconnector v0.97.0
```
Comment (crobert-1) [inline cmd/otelcontribcol/go.mod:12]:
```suggestion
```
The routing connector is already added as required a few lines below.

Comment (akats7) [reply cmd/otelcontribcol/go.mod:12]: Ah yep thanks that shouldn’t be there. Will ipdate

**Comment by djaglowski** [comment]
Thanks for splitting out the bug fix. That one is merged.

File: connector/failoverconnector/metadata.yaml [hunk 2]
```diff
@@ -4,11 +4,11 @@ scope_name: otelcol/failover
 status:
   class: connector
   stability:
-    development: [metrics_to_metrics, traces_to_traces, logs_to_logs]
-  distributions: []
+    alpha: [traces_to_traces, metrics_to_metrics, logs_to_logs]
```
Comment (crobert-1) [inline connector/failoverconnector/metadata.yaml:7]: You'll need to run `make generate` to make sure the component's README also gets updated.

Comment (akats7) [reply connector/failoverconnector/metadata.yaml:7]: Ah good call thanks

**Comment by crobert-1** [comment]
Failing CI/CD :
```
Generated code is out of date, please run "make genotelcontribcol" and commit the changes in this PR.
```

**Review by djaglowski — APPROVED** [review]

**Pull request merged**